### PR TITLE
[video_player] Update iOS/macOS to Pigeon 13

### PR DESCRIPTION
**Pull request by stuartmorgan**
Picks up several breaking changes, including the switch to unwrapped BOOL properties.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m [hunk 1]
```diff
@@ -512,7 +512,7 @@ - (int64_t)duration {
   return FVPCMTimeToMillis([[[_player currentItem] asset] duration]);
 }
 
-- (void)seekTo:(int)location completionHandler:(void (^)(BOOL))completionHandler {
+- (void)seekTo:(int64_t)location completionHandler:(void (^)(BOOL))completionHandler {
```
Comment (stuartmorgan) [inline packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m:515]: We were calling this with `input.position.intValue` previously, which when I removed `intValue` to adjust for the lack of boxing turned into a warning about loss of precision passing an `NSInteger` as an `int`. But this is just used for `CMTimeMake`, which takes an `int64_t`, so it was better to change this type than to pointless cast at the call site and (in theory, but probably never in practice) truncate the value.

File: packages/video_player/video_player_avfoundation/pigeons/messages.dart [hunk 2]
```diff
@@ -7,8 +7,8 @@ import 'package:pigeon/pigeon.dart';
 @ConfigurePigeon(PigeonOptions(
   dartOut: 'lib/src/messages.g.dart',
   dartTestOut: 'test/test_api.g.dart',
-  objcHeaderOut: 'ios/Classes/messages.g.h',
-  objcSourceOut: 'ios/Classes/messages.g.m',
+  objcHeaderOut: 'darwin/Classes/messages.g.h',
+  objcSourceOut: 'darwin/Classes/messages.g.m',
```
Comment (stuartmorgan) [inline packages/video_player/video_player_avfoundation/pigeons/messages.dart:11]: I missed this when I move the files to add macOS support.

File: packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m [hunk 3]
```diff
@@ -709,7 +709,7 @@ - (FVPTextureMessage *)onPlayerSetup:(FVPVideoPlayer *)player
   [eventChannel setStreamHandler:player];
   player.eventChannel = eventChannel;
   self.playersByTextureId[@(textureId)] = player;
-  FVPTextureMessage *result = [FVPTextureMessage makeWithTextureId:@(textureId)];
+  FVPTextureMessage *result = [FVPTextureMessage makeWithTextureId:textureId];
```
Comment (hellohuanlin) [inline packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m:712]: is it pigeon's change? the new pigeon supports primitives and the old version does not?

**Review by hellohuanlin — APPROVED** [review]

File: packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m (hunk 3 above)
Comment (stuartmorgan) [reply packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m:712]: Yes, I changed Pigeon's Obj-C generator to stop boxing non-nullable values, to make them more idiomatic and less error-prone (given that it turns out Xcode is less reliable about warning for buggy implicit `NSNumber*`->`BOOL` conversions than I thought).

File: packages/video_player/video_player_avfoundation/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 2.5.1
+
+* Updates to  Pigeon 13.
```
Comment (tarrinneal) [inline packages/video_player/video_player_avfoundation/CHANGELOG.md:3]: nit '  ' -> ' '.

I can't believe this got merged...

Comment (stuartmorgan) [reply packages/video_player/video_player_avfoundation/CHANGELOG.md:3]: Doh! Luckily it shouldn't show up that way on pub.dev.

**Pull request merged**